src/middlewared/middlewared/utils/origin.py

```diff
@@ -55,16 +61,23 @@ def create(cls, request):
                     pid=pid,
                     uid=uid,
                     loginuid=login_uid,
-                    gid=gid
+                    gid=gid,
+                    secure_transport=True
                 )
             elif sock.family in (AF_INET, AF_INET6):
-                la, lp, ra, rp = get_tcp_ip_info(sock, request)
+                la, lp, ra, rp, ssl = get_tcp_ip_info(sock, request)
+                try:
+                    secure = ssl or ip_address(ra).is_loopback
```

Comment:
Does HA p2p connection play into this at all?
Do we also narrow the secure boolean to the destination port as well?

These may not apply so cordially ignore them if so 🙂